libs/barista-components/autocomplete/src/autocomplete.ts

```diff
@@ -220,6 +226,18 @@ export class DtAutocomplete<T>
 
   ngAfterViewInit(): void {
     this._portal = new TemplatePortal(this._template, this._viewContainerRef);
+    this._options.changes
+      .pipe(
+        takeUntil(this._destroy$),
```

Comment:
The `takeUntil` has to be the last operator – when you perform a `switchMap` you switch your stream to a new source and the new source does not get completed with the takeUntil. 
In your code the `takeUntil` only handles the `option.changes` subscription that is triggered by the `switchMap`.

In your case I would go with a simple store the result as subscription and unsubscribe no need for an extra subject.